src/bin/cargo-flamegraph.rs

```diff
@@ -195,6 +195,20 @@ fn build(opt: &Opt) {
 
         let mut has_debuginfo = false;
 
+        // get names of binaries in the workload
+        let workload = workload(opt)
+            .iter()
+            .map(|w| {
+                w.parse::<PathBuf>()
+                    .unwrap()
+                    .file_name()
+                    .unwrap()
+                    .to_str()
+                    .unwrap()
+                    .to_owned()
```

Comment:
So I'm still a little bit stuck on this thing... Can we do something less finicky here? It's pretty confusing what's going on here, with a lot of unwrapping. Could we make this `filter_map()` and use lots of `ok()?` to just ignore the failures? I think we could do `PathBuf::from(w)` instead of `w.parse::<PathBuf>().unwrap()`?

Reply:
>  think we could do `PathBuf::from(w)`

Yes, good point.

> Could we make this `filter_map()` and use lots of `ok()?`

Yeah, makes sense, would more reliable. I am not exactly sure if this could be made simpler, we are going to use all of these operations anyway to get the file name.